api/applications/applications.proto

```diff
@@ -38,10 +38,22 @@ service Applications {
     }
 }
 
+// This object represents a single condition for a Kubernetes object.
+// It roughly matches the Kubernetes type defined here: https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Condition
+message Condition {
+    string type      = 1;
+    string status    = 2;
+    string reason    = 3;
+    string message   = 4;
+    string timestamp = 5;
+}
+
 message Application {
-    string name = 1;  // The name of the application
-    string path = 2;  // The file path where the k8s yaml files for this application are stored.
-    string url  = 3;  // The git repository URL for this application
+    string   name                            = 1;  // The name of the application
+    string   path                            = 2;  // The file path where the k8s yaml files for this application are stored.
+    string   url                             = 3;  // The git repository URL for this application
+    repeated Condition source_conditions     = 4;  // A list of conditions for the Source related to this Application
+    repeated Condition automation_conditions = 5;  // A list of conditions for the Kustomization or HelmRelease for this application
```

Comment:
the CRD uses the term `deployment` instead of `automation`

Reply:
From another comment that may have shown as "outdated":

> `Deployment` is already a word used in Kubernetes. Do we not already call these things "GOAT" for "GitOps Automation" (or at least that is what I thought it was called)?

So I think we should discuss changing that command-line flag for this reason ^^. That would be outside of the scope of this PR, so I will create a follow up and change this to be consistent for now.

Reply:
Changed to `deployment_conditions`